core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1311,17 +1312,6 @@ class DAGScheduler(
             }
         }
 
-      case Resubmitted =>
```

Comment:
why move the handling of `Resubmitted` after `FetchFailure`?

Reply:
We may have a barrier task failed with reason Resubmitted, so `case Resubmitted` must be after `case failure: TaskFailedReason if task.isBarrier`, and we must make sure that the task failed reasons (except for FetchFailed) are handled inside the barrier task failure handling logic.